### PR TITLE
rename mkl component to bigdl.math

### PR DESCRIPTION
**Pull request by Le-Zheng**
Shall we rename: 
`mkl-java-x86_64-linux` -> `bigdl-core-math-x86_64-linux` 
`mkl-java-mac` -> `bigdl-core-math-mac`

### CODE REVIEW

File: dist/dist-linux/pom.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
   <dependencies>
     <dependency>
       <groupId>com.intel.analytics.bigdl.core.native.mkl</groupId>
```
Comment (qiyuangong) [inline dist/dist-linux/pom.xml:16]: Do we need to change package name?

Comment (jason-dai) [reply dist/dist-linux/pom.xml:16]: yes, change the `groupId`

Comment (glorysdj) [reply dist/dist-linux/pom.xml:16]: groupid better to be unified as bigdl  -> com.intel.analytics.bigdl
artifactId can be bigdl-core-math-x86_64-linux, bigdl-core-math-mac, etc.

Comment (Le-Zheng) [reply dist/dist-linux/pom.xml]: changed artifact ID `com.intel.analytics.bigdl.core.native.mkl` to `com.intel.analytics.bigdl.core.native.math`
https://oss.sonatype.org/content/repositories/snapshots/com/intel/analytics/bigdl/core/native/

File: mkl2017-xeon-blas/pom.xml [hunk 2]
```diff
@@ -56,8 +56,8 @@
     <profile>
       <id>deploy</id>
       <modules>
-        <module>mkl-java-x86_64-linux</module>
-        <module>mkl-java-mac</module>
+        <module>bigdl-core-math-x86_64-linux</module>
+        <module>bigdl-core-math-mac</module>
         <!--<module>mkl-java-win64</module>-->
         <module>mkl-java-rh5</module>
```
Comment (qiyuangong) [inline mkl2017-xeon-blas/pom.xml:62]: Need to change rh5

**Comment by Le-Zheng** [comment]
Build on linux pass:
image
Build on mac pass:
image

File: mkl2017-xeon-blas/bigdl-core-math-mac/pom.xml [hunk 3]
```diff
@@ -11,7 +11,7 @@
   <modelVersion>4.0.0</modelVersion>
 
   <groupId>com.intel.analytics.bigdl.core.native.mkl</groupId>
```
Comment (qiyuangong) [inline mkl2017-xeon-blas/bigdl-core-math-mac/pom.xml:13]: com.intel.analytics.bigdl.core.native.math

File: mkl2017-xeon-blas/pom.xml [hunk 4]
```diff
@@ -25,7 +25,7 @@
     <profile>
```
Comment (qiyuangong) [inline mkl2017-xeon-blas/pom.xml:25]: line 13 <artifactId>mkl-parent</artifactId> also need change

**Review by qiyuangong — APPROVED** [review]
LGTM

**Pull request merged**